CONTRIBUTING.md

```diff
@@ -15,6 +15,46 @@ talk to us! You can find out our thoughts and ensure that your contribution
 won't clash or be obviated by OpenBao's normal direction. A great way to do this
 is via the [GitHub Discussions][2].
 
+## Source-Available License Policy
+
+This policy prohibits the utilization or copy from any source that is protected 
+by a source-available license agreement including Business Source License 1.1. 
+This specifically includes all source repository members under the Hashicorp GitHub 
```

Comment:
Including or excluding the MPL ones? We still depend on a number of HashiCorp libraries; OpenTofu has opted not to fork these MPL licensed projects until upstream is unwilling to take changes, and IMO, it seems like a sane course of action.

Reply:
Just the BSL ones, MPL is not source-available its open-source. Though its easier to say for the common person to just not use the Hashicorp source. This tries to kind of cut both ways. To cover source-available licenses, but to also acknowledge OpenBao's upstream.

Reply:
It is also fitting that the DCO section immediately follow this, as both sections are complementary.

Reply:
Ah, I think you might've dropped the "available" from this sentence? I see source repositories, which I take to be all code repos?

Reply:
I was using source to mean more than just code it could be documentation, media, images, etc. It might be better just to drop the word "source" from that sentence specifically for clarity to refer to all repos generally.

Reply:
Hmmm, I think it'd still read clearer like:

```suggestion
This specifically includes all source-available repositories under the HashiCorp GitHub 
```

as otherwise, a statement like:

> This specifically includes **all** repositories under the HashiCorp GitHub organization.

reads like excluding the MPL-licensed repositories (raft, ... &c) as well (I think, for me, its the _all_ that does it). Which, isn't what you're trying to say :-) 


My 2c.

Reply:
I disagree, its more that we are explicitly including _all_ Hashicorp repos in the policy which is the point and LF point. This policy would then apply at scale large/small depending on how a BUSL license was deployed in a specific repo, while also covering all repos should they change broadly over time.

Reply:
The MPL licensed repos like raft would be fine to consume under this policy. It would only warn the contributor to stop and check licenses first.

Reply:
That language comes from LF, hands are tied on that one.

Reply:
Outside of additional maintainership burdens its probably in our combined best interests to do so long term.